src/material/datepicker/datepicker-content.scss

```diff
@@ -6,6 +6,10 @@ $non-touch-calendar-width:
 // (largest the calendar will get).
 $non-touch-calendar-height: 354px;
 
+// Height should be auto, when the custom header is provided.
+// This will prevent the content from overflowing.
+$non-touch-custom-header-calendar-height: auto;
```

Comment:
I don't think that this needs a separate variable since this file isn't meant to be imported. You can put the `auto` inline instead.